PBTA Histologies: All subtyping gsea rerun (6 of N)

kgaonkar6 opened the pull request:
<!--Hi there, thanks for your contribution! Please take a moment to fill out this template to facilitate the review of your pull request.-->

### Purpose/implementation Section

#### What scientific question is your analysis addressing?
Rna summary modules needed for subtyping need to be updated because of the 8 new stranded files. In this PR re-run `gene-set-enrichment-analysis` used in EPN subtyping , other RNA summary file generation have open PRs and described in #861.

All subtyping modules are also being re-run with pbta-histologies-base.tsv; path have changed to relative analyses/[module]/results paths to access most up-to-date summary files.  All RNA summary and subtyping were run as described in : scripts/run-for-subtyping.sh  

| Module |  Reason |Brief Description | output
|--------|----------|--------|------
| [`gene-set-enrichment-analysis`](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/gene-set-enrichment-analysis) |  adding 8 samples #749 and used in subtyping  | Updated gene set enrichment analysis with appropriate RNA-seq expression data | `results/gsva_scores_stranded.tsv` <br> `results/gsva_scores_polya.tsv` <br> for stranded, polya expression data respectively


**Subtyping modules** 
| Module |  Reason |Brief Description | output
|--------|----------|--------|------
| [`molecular-subtyping-CRANIO`](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/molecular-subtyping-CRANIO) | to add subtyping to base histology  | Molecular subtyping of craniopharyngiomas samples  | `results/CRANIO_molecular_subtype.tsv`
| [`molecular-subtyping-EPN`](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/molecular-subtyping-EPN) | to add subtyping to base histology  | Molecular subtyping of ependymoma tumors | `results/EPN_all_data_withsubgroup.tsv`
| [`molecular-subtyping-EWS`](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/molecular-subtyping-EWS) | to add subtyping to base histology | Reclassification of tumors based on the presence of defining fusions for Ewing Sarcoma per | `results/EWS_samples.tsv`
| [`molecular-subtyping-HGG`](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/molecular-subtyping-HGG) | to add subtyping to base histology  | Molecular subtyping of high-grade glioma samples | `results/HGG_molecular_subtype.tsv`
| [`molecular-subtyping-LGAT`](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/molecular-subtyping-LGAT)| to add subtyping to base histology | Molecular subtyping of Low-grade astrocytic tumor samples  | `results/lgat_subtyping.tsv`
| [`molecular-subtyping-MB`](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/molecular-subtyping-MB) | to add subtyping to base histology  | Molecular classification of Medulloblastoma subtypes | `results/MB_molecular_subtype.tsv` <br> `results/MB_batchcorrected_molecular_subtype.tsv` <br> for uncorrected and batch-corrected input matrix
| [`molecular-subtyping-embryonal`](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/molecular-subtyping-embryonal) | to add subtyping to base histology | Molecular subtyping of non-medulloblastoma, non-ATRT embryonal tumors | `results/embryonal_tumor_molecular_subtypes.tsv`
| [`molecular-subtyping-neurocytoma`](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/molecular-subtyping-neurocytoma) | to add subtyping to base histology | Molecular subtyping of Neurocytoma samples | `results/neurocytoma_subtyping.tsv`
| [`molecular-subtyping-pathology`](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/molecular-subtyping-pathology) | to add subtyping to base histology  | Compile output from other molecular subtyping modules and incorporate pathology feedback  | `results/compiled_molecular_subtyping_with_pathology_feedback.tsv`


#### What was your approach?
Adding condition to use pbta-histologies-base.tsv when running on base histology for subtyping and changing path to relative path to analyses/[module]/results/ instead of data to access updated files. 


#### What GitHub issue does your pull request address?
#861 


### Directions for reviewers. Tell potential reviewers what kind of feedback you are soliciting.

#### Which areas should receive a particularly close look?



#### Is there anything that you want to discuss further?
I ran all the updated modules for the new set of RNA-seq samples as described in scripts/run-for-subtyping.sh. Not sure if that script should be in there but thought I should record the modules re-ran for subtyping for future use.    


#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
Yes


### Results

#### What types of results are included (e.g., table, figure)?
tables


#### What is your summary of the results?

Preliminary checks between `compiled_molecular_subtypes_with_clinical_pathology_feedback.tsv` file in master and `compiled_molecular_subtypes_with_clinical_pathology_feedback.tsv` file from re-running with v18 base-histology are as follows 

```r
# A tibble: 11 x 6
   Kids_First_Partic… sample_id Kids_First_Biospe… molecular_subty… integrated_diagnos… short_histology
   <chr>              <chr>     <chr>              <chr>            <chr>               <chr>          
 1 PT_3AWKWXEV        7316-85   BS_SB12W1XT        HGG, BRAF V600E  High-grade glioma/… HGAT           
 2 PT_8P368R5B        7316-4998 BS_FXJY0MNH        HGG, To be clas… NA                  HGAT           
 3 PT_8VRZRZ7R        7316-536  BS_8QB4S4VA        CNS Embryonal, … CNS Embryonal Tumo… Embryonal Tumor
 4 PT_HSHPZAEZ        7316-545  BS_9CA93S6D        HGG, To be clas… NA                  HGAT           
 5 PT_JT9HH7M6        7316-1455 BS_HE0WJRW6        HGG, To be clas… NA                  HGAT           
 6 PT_RYMG3M91        7316-5812 BS_D7XRFE0R        HGG, To be clas… NA                  HGAT           
 7 PT_W5GP3F6B        7316-5003 BS_KABQQA0T        HGG, To be clas… NA                  HGAT           
 8 PT_X7HC5YCY        7316-255  BS_FN07P04C        HGG, To be clas… NA                  HGAT           
 9 PT_Z4BF2NSB        7316-161  BS_SHJA4MR0        HGG, H3 wildtype High-grade glioma/… HGAT           
10 PT_Z4GS3ZQQ        7316-740  BS_EVCAC3NH        EPN, To be clas… NA                  Ependymoma     
11 PT_Z4GS3ZQQ        7316-740  BS_6ZP2KRPW        EPN, To be clas… NA                  Ependymoma    
```

#### Reproducibility Checklist

<!-- Check all those that apply or remove this section if it is not applicable.-->

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

<!-- Please review https://github.com/AlexsLemonade/OpenPBTA-analysis#documenting-your-analysis -->

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

<!-- IF YOUR PULL REQUEST IS A DATA RELEASE, PLEASE REMOVE THE [HTML COMMENT TAG](https://html.com/tags/comment-tag/) FROM THE SECTION BELOW AND COMPLETE THE CHECKLIST-->

<!-- 
### Data Release Checklist
- [ ] Is the table in doc/data-file-descriptions.md up to date?
- [ ] Is doc/data-format.md up to date?
- [ ] Is doc/release-notes.md up to date?
- [ ] Is download-data.sh up to date?
- [ ] Was download-data.sh tested and did it complete without error?
-->

jharenza commented:
> Preliminary checks between `compiled_molecular_subtypes_with_clinical_pathology_feedback.tsv` file in master and `compiled_molecular_subtypes_with_clinical_pathology_feedback.tsv` file from re-running with v18 base-histology are as follows
> 
> ```r
> # A tibble: 11 x 6
>    Kids_First_Partic… sample_id Kids_First_Biospe… molecular_subty… integrated_diagnos… short_histology
>    <chr>              <chr>     <chr>              <chr>            <chr>               <chr>          
>  1 PT_3AWKWXEV        7316-85   BS_SB12W1XT        HGG, BRAF V600E  High-grade glioma/… HGAT           
>  2 PT_8P368R5B        7316-4998 BS_FXJY0MNH        HGG, To be clas… NA                  HGAT           
>  3 PT_8VRZRZ7R        7316-536  BS_8QB4S4VA        CNS Embryonal, … CNS Embryonal Tumo… Embryonal Tumor
>  4 PT_HSHPZAEZ        7316-545  BS_9CA93S6D        HGG, To be clas… NA                  HGAT           
>  5 PT_JT9HH7M6        7316-1455 BS_HE0WJRW6        HGG, To be clas… NA                  HGAT           
>  6 PT_RYMG3M91        7316-5812 BS_D7XRFE0R        HGG, To be clas… NA                  HGAT           
>  7 PT_W5GP3F6B        7316-5003 BS_KABQQA0T        HGG, To be clas… NA                  HGAT           
>  8 PT_X7HC5YCY        7316-255  BS_FN07P04C        HGG, To be clas… NA                  HGAT           
>  9 PT_Z4BF2NSB        7316-161  BS_SHJA4MR0        HGG, H3 wildtype High-grade glioma/… HGAT           
> 10 PT_Z4GS3ZQQ        7316-740  BS_EVCAC3NH        EPN, To be clas… NA                  Ependymoma     
> 11 PT_Z4GS3ZQQ        7316-740  BS_6ZP2KRPW        EPN, To be clas… NA                  Ependymoma    
> ```

I am not sure what you are saying this table shows - are these similarities/differences/what should we be checking here?

jharenza reviewed (commented):
I think this looks good, but I had a few comments/questions:

1. See my first comment - was there something we should look at closer in the clinical feedback results?
2. It wasn't immediately clear to me in which `scripts` folder you put the `run-for-subtyping.sh` file, but I see it now in the main repo `script` folder. I think we will have to add information about this into the main repo README. 
3. For @jaclyn-taroni - a note that samples that are annotated as `To be classified` have a blank integrated diagnosis for now. (This was done in the `molecular-subtyping-pathology` module in #854 when I added integrated diagnosis/broad and short hist). Once we take care of #748, that part of the pathology module would be removed.

kgaonkar6 commented:
> > Preliminary checks between `compiled_molecular_subtypes_with_clinical_pathology_feedback.tsv` file in master and `compiled_molecular_subtypes_with_clinical_pathology_feedback.tsv` file from re-running with v18 base-histology are as follows
> > ```r
> > # A tibble: 11 x 6
> >    Kids_First_Partic… sample_id Kids_First_Biospe… molecular_subty… integrated_diagnos… short_histology
> >    <chr>              <chr>     <chr>              <chr>            <chr>               <chr>          
> >  1 PT_3AWKWXEV        7316-85   BS_SB12W1XT        HGG, BRAF V600E  High-grade glioma/… HGAT           
> >  2 PT_8P368R5B        7316-4998 BS_FXJY0MNH        HGG, To be clas… NA                  HGAT           
> >  3 PT_8VRZRZ7R        7316-536  BS_8QB4S4VA        CNS Embryonal, … CNS Embryonal Tumo… Embryonal Tumor
> >  4 PT_HSHPZAEZ        7316-545  BS_9CA93S6D        HGG, To be clas… NA                  HGAT           
> >  5 PT_JT9HH7M6        7316-1455 BS_HE0WJRW6        HGG, To be clas… NA                  HGAT           
> >  6 PT_RYMG3M91        7316-5812 BS_D7XRFE0R        HGG, To be clas… NA                  HGAT           
> >  7 PT_W5GP3F6B        7316-5003 BS_KABQQA0T        HGG, To be clas… NA                  HGAT           
> >  8 PT_X7HC5YCY        7316-255  BS_FN07P04C        HGG, To be clas… NA                  HGAT           
> >  9 PT_Z4BF2NSB        7316-161  BS_SHJA4MR0        HGG, H3 wildtype High-grade glioma/… HGAT           
> > 10 PT_Z4GS3ZQQ        7316-740  BS_EVCAC3NH        EPN, To be clas… NA                  Ependymoma     
> > 11 PT_Z4GS3ZQQ        7316-740  BS_6ZP2KRPW        EPN, To be clas… NA                  Ependymoma    
> > ```
> 
> I am not sure what you are saying this table shows - are these similarities/differences/what should we be checking here?

Thanks for the review @jharenza we need to verify if these changes are expected. In the table above are samples that have molecular subtype different from running with v18 base histology compared to OpenPBTA repo. 

Of these 11 the following 8 were added to v18 so it's expected to see these come up as different/additional subtyping results 
BS_FXJY0MNH
BS_KABQQA0T
BS_D7XRFE0R
BS_SHJA4MR0
BS_HE0WJRW6
BS_8QB4S4VA
BS_FN07P04C
BS_SB12W1XT

However,  BS_9CA93S6D which is a HGAT only RNAseq is `HGG, To be classified` in v18 base histology run but in OpenPBTA master it doesn't get assigned any value and is not part of [result](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/molecular-subtyping-HGG/results/HGG_molecular_subtype.tsv) )

Additionally, 7316-740 (RNA-Seq: BS_EVCAC3NH  and WGS: BS_6ZP2KRPW)  were Dysembryoplastic neuroepithelial tumor (DNET) in v17 but now are Ependymoma v18. Things to note is that patient :PT_Z4GS3ZQQ has another sample 7316-1797 is still Dysembryoplastic neuroepithelial tumor (DNET) in v18 base histology release.

jharenza commented:
Thanks for clarifying @kgaonkar6! See my responses below.

> Of these 11 the following 8 were added to v18 so it's expected to see these come up as different/additional subtyping results
> BS_FXJY0MNH
> BS_KABQQA0T
> BS_D7XRFE0R
> BS_SHJA4MR0
> BS_HE0WJRW6
> BS_8QB4S4VA
> BS_FN07P04C
> BS_SB12W1XT

✔️ 

> However, BS_9CA93S6D which is a HGAT only RNAseq is `HGG, To be classified` in v18 base histology run but in OpenPBTA master it doesn't get assigned any value and is not part of [result](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/molecular-subtyping-HGG/results/HGG_molecular_subtype.tsv) )

This sample had `pathology_diagnosis == Not Reported` in V17, but should be captured in the V18 subtyping run - you will see it in your `analyses/molecular-subtyping-HGG/results/HGG_molecular_subtype.tsv` file in line 106, which makes its way to the final file. So, this is expected.

> Additionally, 7316-740 (RNA-Seq: BS_EVCAC3NH and WGS: BS_6ZP2KRPW) were Dysembryoplastic neuroepithelial tumor (DNET) in v17 but now are Ependymoma v18. Things to note is that patient :PT_Z4GS3ZQQ has another sample 7316-1797 is still Dysembryoplastic neuroepithelial tumor (DNET) in v18 base histology release.

Confirmed that 7316-740 (Initial Tumor) was updated to EPN in the latest data refresh and 7316-1797 (Recurrence) is still annotated as DNET.

Thanks for doing these checks! 🦅 👀

jharenza reviewed (approved):
Looks good. Will you also update the main repo README per this comment:

>It wasn't immediately clear to me in which scripts folder you put the run-for-subtyping.sh file, but I see it now in the main repo script folder. I think we will have to add information about this into the main repo README.

kgaonkar6 commented:
> Looks good. Will you also update the main repo README per this comment:
> 
> > It wasn't immediately clear to me in which scripts folder you put the run-for-subtyping.sh file, but I see it now in the main repo script folder. I think we will have to add information about this into the main repo README.
> 
> You can create a new subheading with text similar to the section on creating files for CI.
 
I've updated the main repo README according to  the suggestion above by @jharenza  and analyses/README is now updated to relative paths for RNAseq files in molecular-subtyping* modules.

jharenza commented on README.md:
```diff
@@ -479,4 +481,40 @@ In this example `OPENPBTA_PATHSIG=0.75` species an environment variable `OPENPBT
 
+### Molecular-subtyping 
+
+If you would like to identify molecular subtype membership for new RNAseq PBTA samples belonging to the following broad_histology
```
```suggestion
If you would like to identify molecular subtype membership for new RNAseq PBTA samples belonging to the following broad_histologies, run the bash script below.
```

jharenza commented on README.md:
````diff
@@ -479,4 +481,40 @@ In this example `OPENPBTA_PATHSIG=0.75` species an environment variable `OPENPBT
+bash scripts/run-for-subtyping.sh
+```
+
+Running this will re-run RNA-seq specific summary file generation modules as well as molecular-subtyping-* modules to generate a `compiled_molecular_subtypes_with_clinical_pathology_feedback.tsv` with molecular_subtype which is updated by pathology/clinical feedback information when available.
````
```suggestion
Running this will re-run RNA-seq specific summary file generation modules as well as molecular-subtyping-* modules to generate the `compiled_molecular_subtypes_with_clinical_pathology_feedback.tsv` file containing the `molecular_subtype` column.
```

jharenza commented on README.md:
```diff
@@ -479,4 +481,40 @@ In this example `OPENPBTA_PATHSIG=0.75` species an environment variable `OPENPBT
+
+#### Adding summary analyses to run-for-subtyping.sh
+
+For an analysis to be run for subyping, it must use pbta-histologies-base.tsv and shouldn't depend on `molecular_subtype` or `integrated_diagnosis` column annotation to generate files which are inputs for molecular-subtyping-* modules. Please set BASE_SUBTYPING=1 as a condition to run code with `pbta-histologies-base.tsv`.   
```
```suggestion
For an analysis to be run for subyping, it must use `pbta-histologies-base.tsv` as input and shouldn't depend on `molecular_subtype` or `integrated_diagnosis` columns for molecular-subtyping-* modules. 
Please set BASE_SUBTYPING=1 as a condition to run code with `pbta-histologies-base.tsv`.   
```

jharenza commented on README.md:
```diff
@@ -479,4 +481,40 @@ In this example `OPENPBTA_PATHSIG=0.75` species an environment variable `OPENPBT
+
+For an analysis to be run for subyping, it must use pbta-histologies-base.tsv and shouldn't depend on `molecular_subtype` or `integrated_diagnosis` column annotation to generate files which are inputs for molecular-subtyping-* modules. Please set BASE_SUBTYPING=1 as a condition to run code with `pbta-histologies-base.tsv`.   
+
+Here is an example 
```
```suggestion
Here is an example:
```

jharenza commented on README.md:
````diff
@@ -479,4 +481,40 @@ In this example `OPENPBTA_PATHSIG=0.75` species an environment variable `OPENPBT
+
+```
+
+This would run the `analyses/gene-set-enrichment-analysis/run-gsea.sh` with pbta-histologies-base.tsv to generate gsva scores that are used in multiple molecular-subtyping-* modules.
````
```suggestion
This would run the `analyses/gene-set-enrichment-analysis/run-gsea.sh` with `pbta-histologies-base.tsv` to generate gsva scores that are used in multiple molecular-subtyping-* modules.
```

jharenza reviewed (commented):
Just have a few minor updates to the README text, and then I think this one is good to go!

kgaonkar6 commented:
Latest commit changes was from a re-run with updated base histology(minor change to Normal sample rows and CNS_region updates) file which causes the difference in CNS_region column in subsetted molecular-subtyping metadata files.

cansavvy commented:
Ignoring the transcriptomic changes from #868 , this looks mostly on track, but I'm a bit confused by we are seeing `fusion-filtering` changes even after we've merged to master? I think this is a thing we saw previously: https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/866#discussion_r552691284 
It's unclear to me why it's happening again.

cansavvy commented:
I'm going to run a `git checkout master analyses/fusion-filtering/*` to revert the fusion-filtering changes that I don't think belong here and make sure those files are concurrent with master. It may have been a merge conflict gone wrong. ¯\\_(ツ)_/¯

cansavvy commented:
> which seems mainly line 109 which has the condition to run for subtyping

Can you link to what file you are talking about here? 

> I should revert these files to master.

Which files should be reverted specifically? 

Would this be resolved by just re-running the `scripts/run-for-subtyping.sh` ?

kgaonkar6 commented:
git checkout master analyses/fusion-filtering/* fixes it but just FYI these files will come up in #870 as well where they need to be reverted back to master.


the changes come in file: https://github.com/kgaonkar6/OpenPBTA-analysis/blob/v18_int_dx_histology/analyses/fusion_filtering/run_fusion_merged.sh

because as you see in this file from #870 there is a chunk of code for if [ "$RUN_FOR_SUBTYPING" == 1 ] and another for if [ "$RUN_FOR_SUBTYPING" == 0] but then I updated the code from your [comment](https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/865#discussion_r545097390) I updated to specify a variable histology_file for each condition instead of duplicating the code as bellow:
https://github.com/kgaonkar6/OpenPBTA-analysis/blob/24486f87980be9aeff3fe920f4c346d2465eb34c/analyses/fusion_filtering/run_fusion_merged.sh#L48-L56
 

This was useful to make the code readable but I forgot to carry over these changes to the remaining branches so they come up as changes in each branch staggered from the original fusion_filtering_base.

cansavvy commented:
> This was useful to make the code readable but I forgot to carry over these changes to the remaining branches so they come up as changes in each branch staggered from the original fusion_filtering_base branch.

Gotcha yes, that's the tricky part. 

So do we otherwise think this is resolved here? (And we can use `git checkout master analyses/fusion-filtering/* ` in other PRs where needed if this did indeed solve the problem).

kgaonkar6 commented:
Yes, I looks like it's resolved here and doing the same in #870 will work as well .

jaclyn-taroni commented:
When I fix these merge conflicts ☝️ locally, I'm going to select the recent the most commit to master (https://github.com/AlexsLemonade/OpenPBTA-analysis/commit/68cf73d22cd1c5224a451c65142f5a1aa6ac88f7).

cansavvy commented:
Hi @kgaonkar6 , we were trying to parse apart some of these merge conflicts and are noticing that some logic in dimension-reduction-plots.sh was lost in #868 in this particular commit: 

https://github.com/kgaonkar6/OpenPBTA-analysis/commit/ee99f3b1fbb2f68013dcffa3d5a4eb09a076662e#diff-7589ddb81e105a0690ef506affa80ac22bd110dbde2ba467cd0e74a2749e4862

But this logic is here in this PR. Was dropping this if statement linked above intentional? I think we want this logic, right? If yes, then I will make sure its added back in the merge conflict here.

kgaonkar6 commented:
kgaonkar6@ee99f3b#diff-7589ddb81e105a0690ef506affa80ac22bd110dbde2ba467cd0e74a2749e4862 was intentional 

But reviewing the master scripts for tsne I think something got lost either by me or while conflict resolutions maybe?

The 01-dimension-reduction.sh should have a section to specify METADATA as per running suubtyping or not:
https://github.com/kgaonkar6/OpenPBTA-analysis/blob/24486f87980be9aeff3fe920f4c346d2465eb34c/analyses/transcriptomic-dimension-reduction/01-dimension-reduction.sh#L19-L29

the file in master doesn't have those lines https://github.com/kgaonkar6/OpenPBTA-analysis/blob/master/analyses/transcriptomic-dimension-reduction/01-dimension-reduction.sh.

cansavvy commented:
Note that the comment above is about `dimension-reduction-plots.sh` NOT `01-dimension-reduction.sh` which is what you are referring to here.

kgaonkar6 commented:
> Hi @kgaonkar6 , we were trying to parse apart some of these merge conflicts and are noticing that some logic in dimension-reduction-plots.sh was lost in #868 in this particular commit:
> 
> [kgaonkar6@ee99f3b#diff-7589ddb81e105a0690ef506affa80ac22bd110dbde2ba467cd0e74a2749e4862](https://github.com/kgaonkar6/OpenPBTA-analysis/commit/ee99f3b1fbb2f68013dcffa3d5a4eb09a076662e#diff-7589ddb81e105a0690ef506affa80ac22bd110dbde2ba467cd0e74a2749e4862)
> 
> But this logic is here in this PR. Was dropping this if statement linked above intentional? I think we want this logic, right? If yes, then I will make sure its added back in the merge conflict here.


removing if [ "$RUN_FOR_SUBTYPING" == 0 ]; then from dimension-reduction-plots.sh was intentional. Because I had added that condition to 01-dimension-reduction.sh only since only   01-dimension-reduction.sh  uses the METADATA information.

cansavvy commented:
> removing if [ "$RUN_FOR_SUBTYPING" == 0 ]; then from dimension-reduction-plots.sh was intentional. Because I had added that condition to 01-dimension-reduction.sh only since only 01-dimension-reduction.sh uses the METADATA information.

@kgaonkar6  Gotcha. So my guess is that the variable setting should have also been removed at that time since its no longer being used: https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/f63cdb824af4caeca8dd598db63a8ca29046bf96/analyses/transcriptomic-dimension-reduction/dimension-reduction-plots.sh#L10

Is this correct? If so I'll push a commit that removes that.

cansavvy commented on analyses/gene-set-enrichment-analysis/run-gsea.sh:
```diff
@@ -18,6 +20,8 @@ set -o pipefail
 
 IS_CI=${OPENPBTA_TESTING:-0}
+RUN_FOR_SUBTYPING=${OPENPBTA_BASE_SUBTYPING:-0}
+echo $RUN_FOR_SUBTYPING
```
This feels like it might be a development remnant, but it also doesn't hurt anything. ¯\\_(ツ)_/¯

kgaonkar6 replied:
it's intentional since with the condition we can run gsea with DATA_DIR="../collapse-rnaseq/results" which is updated as part of subtyping https://github.com/kgaonkar6/OpenPBTA-analysis/blob/all_subtyping_gsea_rerun/scripts/run-for-subtyping.sh

cansavvy replied:
I was specifically referring to the `echo` line. Since I don't see that elsewhere.

cansavvy commented on analyses/gene-set-enrichment-analysis/run-gsea.sh:
```diff
@@ -40,6 +45,22 @@ INPUT_FILE="${DATA_DIR}/pbta-gene-expression-rsem-fpkm-collapsed.stranded.rds"
+# Only run when pbta-histologies.tsv is generated which has integrated_diagnosis
 Rscript -e "rmarkdown::render('02-model-gsea.Rmd', clean = TRUE, params=list(is_ci = ${IS_CI}))"
+else
+DATA_DIR="../collapse-rnaseq/results"
```
I *think the `DATA_DIR` and `RESULTS_DIR` bits are the only parts that change in this if/then, else statement. If so, this can be made a lot shorter by only having these `DATA_DIR` and `RESULTS_DIR` declarations in the if statement and you only need the actual script calls once.

kgaonkar6 replied:
oh yeah sounds good, I'll update that

kgaonkar6 replied:
ahh I didn't read my own comment :D so the DATA_DIR and RESULTS_DIR bits change as per the condition for subtyping is T/F. but also 02-model-gsea.Rmd only needs to be run when we have the final pbta-histologies.tsv. Which also brings up the #877 issue that this needs to be harmonized_diagnosis in a future PR.

```
# Only run when pbta-histologies.tsv is generated which has integrated_diagnosis
Rscript -e "rmarkdown::render('02-model-gsea.Rmd', clean = TRUE, params=list(is_ci = ${IS_CI}))"
```

cansavvy replied:
Okay, but aren't lines 37 - 46 and 55 - 64  and still redundant? I think you can include this rmarkdown call in the T/F statement but still don't need the rest of this repeated.

cansavvy commented on analyses/molecular-subtyping-HGG/02-HGG-molecular-subtyping-subset-files.R:
```diff
@@ -71,14 +77,14 @@ cn_df <- read_tsv(file.path(
 # Read in fusion data
 fusion_df <- read_tsv(
-  file.path(data_dir, "pbta-fusion-putative-oncogenic.tsv"))
+  file.path(root_dir, "analyses","fusion_filtering", "results", "pbta-fusion-putative-oncogenic.tsv"))
```
Is there a particular reason you didn't change the `data_dir` specification but instead changed the file path here? I noticed this happens a few times in this PR.

kgaonkar6 replied:
since the relative analyses/fusion_filtering/results would have the updated files from https://github.com/kgaonkar6/OpenPBTA-analysis/blob/all_subtyping_gsea_rerun/scripts/run-for-subtyping.sh I pointed to thatt folder instead of data_dir. Also data_dir is used for other files as well so didn't change that variable.

cansavvy replied:
Okay. No problem. To keep it parallel, you could specify two different data_dir and call one collapse_data_dir, or whatever is appropriate, but this seems fine for the time being.

cansavvy commented on analyses/molecular-subtyping-HGG/02-HGG-molecular-subtyping-subset-files.R:
```diff
@@ -47,15 +47,21 @@ select_metadata <- metadata %>%
   read_rds(
     file.path(
-      data_dir,
+      root_dir,
```
Here's another instance of what I mentioned previously about `data_dir`.

kgaonkar6 replied:
This seems like the same as above : since the relative analyses/collapse-rnaseq/results/  would have the updated files from https://github.com/kgaonkar6/OpenPBTA-analysis/blob/all_subtyping_gsea_rerun/scripts/run-for-subtyping.sh I pointed to that folder instead of data_dir.

cansavvy commented on analyses/molecular-subtyping-pathology/01-compile-subtyping-results.Rmd:
```diff
@@ -290,7 +284,6 @@ mb_results_df <- bind_exp_strategies(mb_results_df) %>%
 cranio_results_df <- bind_exp_strategies(cranio_results_df) %>%
   inner_join(select(histologies_df,
                     Kids_First_Biospecimen_ID,
-                    integrated_diagnosis,
```
Is this an instance where #877 applies? We don't need to change this now necessarily. Just wondering for future reference.

kgaonkar6 replied:
hmm good point, I think #877 doesn't apply to this step since in molecular-subtyping-integrate if a sample under goes OpenPBTA subtyping the `integrated_diagnosis` is used as `harmonized_daiagnosis`.

kgaonkar6 replied:
also just wanted to note here that since in pbta-histologies-base.tsv `integrated_diagnosis` == NA so this column was removed to be able to directly add the integrated_diagnosis from the subtyping results.

cansavvy reviewed (commented):
I have only a couple comments. The results files I've just skimmed through and I'm assuming they make sense.

But otherwise, I think this is nearly ready for merging.

kgaonkar6 commented:
I just updated the circleci/config.yml file which was erroring out at GSEA as well , which might help pass CI.

cansavvy commented:
@kgaonkar6 This commit: https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/869/commits/f1b36339ec435eef20c27fa5b11892d203723699 shows 61 files changed. Can you explain what this was for? It's unclear to me looking at it.

kgaonkar6 commented:
that was from git pull command to update my local branch to remote `all_subtyping_gsea_rerun`

[git_pull_61_files.txt](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/5783712/git_pull_61_files.txt)

cansavvy reviewed (approved):
Okay. Thanks that print out was helpful. The files that were changed according to that git print out file you sent are not showing up in the `files changed` here, which means that they concurrent with what is in `master`. So now that that is settled and the other changes have been addressed. I think this is all set to go in assuming CI passes (which I expect it to). :+1:

cansavvy commented:
> I copied this from my terminal, wish github would show the files added from `git merge` commands .. easier to look into .. maybe there is a way I don't know

As far as this comment: https://stackoverflow.com/questions/424071/how-to-list-all-the-files-in-a-commit You can try this out. Perhaps it's easier than what you had to do to get that `git_pull_61_files.txt`?

kgaonkar6 commented:
ah thanks! this worked perfectly 
`git show --stat f1b3633`
